src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -733,4 +855,263 @@ export class NetworkAreaDiagramViewer {
             points[1].y.toFixed(2);
         polyline?.setAttribute('points', polylinePoints);
     }
+
+    private moveLoopEdgeGroup(edges: SVGGraphicsElement[], offset: Point) {
+        edges.forEach((edge) => {
+            // get edge element
+            const edgeNode: SVGGraphicsElement | null =
+                this.container.querySelector(
+                    "[id='" + edge.getAttribute('svgid') + "']"
+                );
+            if (!edgeNode) {
+                return;
+            }
+            const translation = new Point(
+                offset.x - this.initialPosition.x,
+                offset.y - this.initialPosition.y
+            );
+            const transform = DiagramUtils.getTransform(edgeNode);
+            const totalTranslation = new Point(
+                (transform?.matrix.e ?? 0) + translation.x,
+                (transform?.matrix.f ?? 0) + translation.y
+            );
+            //transform?.setTranslate(totalTranslation.x, totalTranslation.y);
+            edgeNode?.setAttribute(
+                'transform',
+                'translate(' +
+                    totalTranslation.x.toFixed(2) +
+                    ',' +
+                    totalTranslation.y.toFixed(2) +
+                    ')'
+            );
+            // store edge angles, to use them for bus node redrawing
+            this.storeHalfLoopEdgeAngle(edgeNode, edgeNode.id + '.1', 0);
+            this.storeHalfLoopEdgeAngle(edgeNode, edgeNode.id + '.2', 1);
+        });
+    }
+
+    private storeHalfLoopEdgeAngle(
```

Comment:
Can't you factorize this method with `storeHalfEdgeAngle`, taking a function `getAngle: (a: HTMLElement) => number` as parameter? which would be either `DiagramUtils::getPathAngle` or `DiagramUtils::getPolylineAngle`.

Reply:
getAngle function parameter added to storeHalfEdgeAngle method